### PR TITLE
Add create the Spring example

### PR DESCRIPTION
**Pull request by tkoyama010**
### Overview

Add create the Spring example from https://kitware.github.io/vtk-examples/site/Python/Modelling/Spring/ .
spring_example

 

### Details

- None

### CODE REVIEW

**Review by tkoyama010 — COMMENTED** [review]
WFM

File: examples/01-filter/extrude-rotate.py [hunk 1]
```diff
@@ -44,3 +45,33 @@
 )
 
 plotter.show(cpos="xy")
+
+###############################################################################
+# Create the spring
```
Comment (akaszynski) [inline examples/01-filter/extrude-rotate.py:50]:
```suggestion
# Create a spring
```

File: examples/01-filter/extrude-rotate.py [hunk 2]
```diff
@@ -44,3 +45,33 @@
...
+# ~~~~~~~~~~~~~~~~~
```
Comment (akaszynski) [inline examples/01-filter/extrude-rotate.py:51]:
```suggestion
# Create a spring
# ~~~~~~~~~~~~~~~
```

File: pyvista/core/filters.py [hunk 3]
```diff
@@ -5052,6 +5053,12 @@ def extrude_rotate(poly_data, resolution=30, inplace=False, progress_bar=False):
         inplace : bool, optional
             Overwrites the original mesh inplace.
 
+        translation : float
+            Total amount of translation along the z-axis.
+
+        dradius : float
```
Comment (akaszynski) [inline pyvista/core/filters.py:5059]:
```suggestion
        translation : float, optional
            Total amount of translation along the z-axis.

        dradius : float, optional
```

File: pyvista/core/filters.py [hunk 4]
```diff
@@ -5052,6 +5053,12 @@ def extrude_rotate(poly_data, resolution=30, inplace=False, progress_bar=False):
...
+            Change in radius during sweep process.
+
```
Comment (akaszynski) [inline pyvista/core/filters.py:5061]:
```suggestion
    angle : float, optional
        The angle of rotation.

```

File: examples/01-filter/extrude-rotate.py [hunk 5]
```diff
@@ -44,3 +45,33 @@
...
 
 plotter.show(cpos="xy")
+
+###############################################################################
+# Create a spring
+# ~~~~~~~~~~~~~~~
+
+# Create the spring profile (a circle).
+plotter = pyvista.Plotter(shape=(1, 1))
+plotter.subplot(0, 0)
```
Comment (akaszynski) [inline examples/01-filter/extrude-rotate.py:55]:
```suggestion
plotter = pyvista.Plotter()
```
This is already default.

**Review by akaszynski — APPROVED** [review]
Great example!  Thanks for adding it and improving `extrude_rotate`.

File: examples/01-filter/extrude-rotate.py (hunk 5 above)
Comment (tkoyama010) [reply examples/01-filter/extrude-rotate.py:55]: Thanks

File: pyvista/core/filters.py [hunk 6]
```diff
@@ -5052,6 +5053,15 @@ def extrude_rotate(poly_data, resolution=30, inplace=False, progress_bar=False):
...
 
+        translation : float, optional
+            Total amount of translation along the z-axis.
+
+        dradius : float, optional
+            Change in radius during sweep process.
+
+    angle : float, optional
+        The angle of rotation.
+
```
Comment (tkoyama010) [inline pyvista/core/filters.py:5064]:
```suggestion
        angle : float, optional
            The angle of rotation.

```

File: tests/test_filters.py [hunk 7]
```diff
@@ -1418,6 +1418,21 @@ def test_extrude_rotate():
...
+    assert zmax == translation
+    xmax = poly.bounds[1]
+    assert xmax == line.bounds[1] + dradius
+
+    poly = line.extrude_rotate(angle=90.0)
+    xmin = poly.bounds[0]
+    xmax = poly.bounds[1]
+    ymin = poly.bounds[2]
+    ymax = poly.bounds[3]
+    assert (xmin == 0.0) and (xmax == 1.0) and (ymin == 0.0) and (ymax == 1.0)
```
Comment (tkoyama010) [inline tests/test_filters.py:1434]:
```suggestion
    assert (xmin == line.bounds[0]) and (xmax == line.bounds[1]) and (ymin == line.bounds[0]) and (ymax == line.bounds[1])
```

**Pull request merged**